command/agent/host_volume_endpoint_test.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package agent
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"github.com/hashicorp/nomad/nomad/mock"
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/shoenig/test/must"
+)
+
+func TestHostVolumeEndpoint_CRUD(t *testing.T) {
+	httpTest(t, nil, func(s *TestAgent) {
+
+		// Create a volume on the test node
+
+		vol := mock.HostVolumeRequest()
+		reqBody := struct {
+			Volumes []*structs.HostVolume
+		}{Volumes: []*structs.HostVolume{vol}}
+		buf := encodeReq(reqBody)
+		req, err := http.NewRequest(http.MethodPut, "/v1/volume/host/create", buf)
+		must.NoError(t, err)
+		respW := httptest.NewRecorder()
+
+		// Make the request and verify we got a valid volume back
+
+		obj, err := s.Server.HostVolumeSpecificRequest(respW, req)
+		must.NoError(t, err)
+		must.NotNil(t, obj)
+		resp := obj.(*structs.HostVolumeCreateResponse)
+		must.Len(t, 1, resp.Volumes)
+		must.Eq(t, vol.Name, resp.Volumes[0].Name)
+		must.Eq(t, s.client.NodeID(), resp.Volumes[0].NodeID)
+		must.NotEq(t, "", respW.HeaderMap.Get("X-Nomad-Index"))
+
+		volID := resp.Volumes[0].ID
+
+		// Verify volume was created
+
+		path, err := url.JoinPath("/v1/volume/host/", volID)
+		must.NoError(t, err)
+		req, err = http.NewRequest(http.MethodGet, path, nil)
+		must.NoError(t, err)
+		obj, err = s.Server.HostVolumeSpecificRequest(respW, req)
+		must.NoError(t, err)
+		must.NotNil(t, obj)
+		respVol := obj.(*structs.HostVolume)
+		must.Eq(t, s.client.NodeID(), respVol.NodeID)
+
+		// Update the volume (note: this doesn't update the volume on the client)
```

Comment:
pardon the tangent, but this parenthetical made me second-guess Create RPC.  I think right now repeat Create calls with the same `ID`, but no `NodeID`, get "scheduled" to a(nother) node, because of this check: https://github.com/hashicorp/nomad/blob/da738e70cf51294c83f8c8b1cdf4e8137eb0fdcd/nomad/host_volume_endpoint.go#L377

if that's the case, multiple Nodes could end up with the "same" volume on them.

our desired behavior would be to either:
* send the Create client RPC to the _current_ node/client, and execute the plugin expecting it to be appropriately idempotent
* do that only if there are any changes to the volume

at any rate, I think we need to determine `NodeID` on `HostVolumeCreateRequest.Volumes` that only have `ID`.